trtools/testsupport/sample_vcfs/compareSTR_vcfs/test_hipstr_flanking_bp_flanking.vcf

```diff
@@ -0,0 +1,41 @@
+##INFO=<ID=DP,Number=1,Type=Integer,Description="Total number of valid reads used to genotype all samples">
+##INFO=<ID=DSNP,Number=1,Type=Integer,Description="Total number of reads with SNP phasing information">
+##INFO=<ID=DSTUTTER,Number=1,Type=Integer,Description="Total number of reads with a stutter indel in the STR region">
+##INFO=<ID=DFLANKINDEL,Number=1,Type=Integer,Description="Total number of reads with an indel in the regions flanking the STR">
+##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">
+##FORMAT=<ID=GB,Number=1,Type=String,Description="Base pair differences of genotype from reference">
+##FORMAT=<ID=Q,Number=1,Type=Float,Description="Posterior probability of unphased genotype">
+##FORMAT=<ID=PQ,Number=1,Type=Float,Description="Posterior probability of phased genotype">
+##FORMAT=<ID=DP,Number=1,Type=Integer,Description="Number of valid reads used for sample's genotype">
+##FORMAT=<ID=DSNP,Number=1,Type=Integer,Description="Number of reads with SNP phasing information">
+##FORMAT=<ID=PSNP,Number=1,Type=String,Description="Number of reads with SNPs supporting each haploid genotype">
+##FORMAT=<ID=PDP,Number=1,Type=String,Description="Fractional reads supporting each haploid genotype">
+##FORMAT=<ID=GLDIFF,Number=1,Type=Float,Description="Difference in likelihood between the reported and next best genotypes">
+##FORMAT=<ID=DSTUTTER,Number=1,Type=Integer,Description="Number of reads with a stutter indel in the STR region">
+##FORMAT=<ID=DFLANKINDEL,Number=1,Type=Integer,Description="Number of reads with an indel in the regions flanking the STR">
+##FORMAT=<ID=AB,Number=1,Type=Float,Description="log10 of the allele bias pvalue, where 0 is no bias and more negative values are increasingly biased. 0 for all homozygous genotypes">
+##FORMAT=<ID=FS,Number=1,Type=Float,Description="log10 of the strand bias pvalue from Fisher's exact test, where 0 is no bias and more negative values are increasingly biased. 0 for all homozygous genotypes">
+##FORMAT=<ID=DAB,Number=1,Type=Integer,Description="Number of reads used in the AB and FS calculations">
+##FORMAT=<ID=ALLREADS,Number=1,Type=String,Description="Base pair difference observed in each read's Needleman-Wunsch alignment">
+##FORMAT=<ID=MALLREADS,Number=1,Type=String,Description="Maximum likelihood bp diff in each read based on haplotype alignments for reads that span the repeat region by at least 5 base pairs">
+##contig=<ID=1,length=249250621>
+#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	GTEX-R55C-0003
+1	101673	STR_38	TCAAAAAAAAAAAAAAAA	TAAAAAAAAAAAAAAAAA	.	.	INFRAME_PGEOM=0.95;INFRAME_UP=0.05;INFRAME_DOWN=0.05;OUTFRAME_PGEOM=0.95;OUTFRAME_UP=0.01;OUTFRAME_DOWN=0.01;BPDIFFS=0;START=101675;END=101690;PERIOD=1;AN=2;REFAC=1;AC=1;NSKIP=0;NFILT=0;DP=53;DSNP=0;DSTUTTER=2;DFLANKINDEL=1	GT:GB:Q:PQ:DP:DSNP:DSTUTTER:DFLANKINDEL:PDP:PSNP:GLDIFF:AB:DAB:FS:ALLREADS:MALLREADS	0|1:0|0:1.0:0.5:53:0:2:1:41.95|11.05:0|0:5.08:-5.03:49:-0.39:-19|1;-15|1;-2|1;0|40:-2|1;0|39;1|1
```

Comment:
I've added the test cases you mentioned, apart from the last one. compareSTR currently compares records that have different start/end bps, it just evaluates their seq concordance to 0. Did you mean test where records have different starting position, and harmonization step isn't able to make them comparable?

Reply:
"Did you mean test where records have different starting position, and harmonization step isn't able to make them comparable?" - yes.

It seems like currently records from different files that overlap but have different starting bps are silently ignored. It would be worthwhile to instead output an error saying something like 'the records <rec1>,<rec2> only partially overlap and so were not compared' (and then write a test to see that that is happening).

And also the flip of that test, taking records from different files which have the same reference starting positions but different reference ending positions. Currently, those records are compared and marked as nonequivalent, it would be better if they were not compared so this was treated the same way as records which had different starting positions.

Reply:
Sounds good.